### PR TITLE
[KOGITO-7700] - Allow adding quarkus extensions when building from

### PR DESCRIPTION
**Pull request by spolti**
assets
See: https://issues.redhat.com/browse/KOGITO-7700

Fixes https://issues.redhat.com/browse/KOGITO-7791

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <!-- Keep them in alphabetical order -->
 ## Enhancements
 [KOGITO-7744](https://issues.redhat.com/browse/KOGITO-7744) - Move ENABLE_RH_MAVEN_REPO from maven module to prod module
+[KOGITO-7700](https://issues.redhat.com/browse/KOGITO-7700) - Allow adding quarkus extensions when building from assets
 
 ## Bug Fixes
+[KOGITO-77910](https://issues.redhat.com/browse/KOGITO-7791) - [TEST] - Serverless Workflow Extension is building the service properly is failing
```
Comment (radtriste) [inline RELEASE_NOTES.md]:
```suggestion
[KOGITO-7791](https://issues.redhat.com/browse/KOGITO-7791) - [TEST] - Serverless Workflow Extension is building the service properly is failing
```

File: modules/kogito-s2i-core/module.yaml [hunk 2]
```diff
@@ -54,6 +54,10 @@ envs:
     description: ^ Defines the Quarkus platform artifact id, defaults to quarkus-bom for community and product.
     example: "quarkus-bom-custom"
 
+  - name: QUARKUS_EXTRA_EXTENSIONS
+    description: ^ Defines the Quarkus' extra extensions to be added when performing build from assets.
```
Comment (radtriste) [inline modules/kogito-s2i-core/module.yaml]:
```suggestion
    description: ^ Defines the Quarkus' extra extensions (as comma-separated list) to be added when performing build from assets.
```

**Review by radtriste — APPROVED** [review]
minor suggestions

**Comment by kie-ci1** [comment]
**PR job** `#61` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/61/display/redirect

**Test results:**
- PASSED: 235
- FAILED: 3

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/61/testReport/features.image.kogito-builder-native/kogito-builder image native build tests/Verify_if_the_s2i_build_is_finished_as_expected_performing_a_native_build_with_persistence_enabled___Step_1:_build_the_application_and_copy_to_the_runtime_image/">features.image.kogito-builder-native.kogito-builder image native build tests.Verify if the s2i build is finished as expected performing a native build with persistence enabled - Step 1: build the application and copy to the runtime image</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/61/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_that_the_Kogito_Quarkus_Serverless_Workflow_Extension_is_building_the_service_properly/">features.image.kogito-builder.kogito-builder image tests.Verify that the Kogito Quarkus Serverless Workflow Extension is building the service properly</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/61/testReport/features.image.kogito-common-builder-jvm/kogito-builder image JVM build tests/Verify_if_the_s2i_build_is_finished_as_expected_with_uber_jar_package_type_built/">features.image.kogito-common-builder-jvm.kogito-builder image JVM build tests.Verify if the s2i build is finished as expected with uber-jar package type built</a></summary>
S2I build failed, check logs!
</details>

**Comment by kie-ci1** [comment]
**PR job** `#63` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/63/display/redirect

**Test results:**
- PASSED: 234
- FAILED: 4

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/63/testReport/features.image.kogito-builder-native/kogito-builder image native build tests/Verify_if_the_s2i_build_is_finished_as_expected_performing_a_native_build_with_persistence_enabled___Step_1:_build_the_application_and_copy_to_the_runtime_image/">features.image.kogito-builder-native.kogito-builder image native build tests.Verify if the s2i build is finished as expected performing a native build with persistence enabled - Step 1: build the application and copy to the runtime image</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/63/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_if_the_s2i_build_is_finished_as_expected_performing_a_non_native_build_with_persistence_enabled/">features.image.kogito-builder.kogito-builder image tests.Verify if the s2i build is finished as expected performing a non native build with persistence enabled</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/63/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_that_the_Kogito_Quarkus_Serverless_Workflow_Extension_is_building_the_service_properly/">features.image.kogito-builder.kogito-builder image tests.Verify that the Kogito Quarkus Serverless Workflow Extension is building the service properly</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/63/testReport/features.image.kogito-common-builder-jvm/kogito-builder image JVM build tests/Verify_if_the_s2i_build_is_finished_as_expected_with_uber_jar_package_type_built/">features.image.kogito-common-builder-jvm.kogito-builder image JVM build tests.Verify if the s2i build is finished as expected with uber-jar package type built</a></summary>
S2I build failed, check logs!
</details>

**Comment by kie-ci1** [comment]
**PR job** `#62` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/62/display/redirect

**Test results:**
- PASSED: 235
- FAILED: 3

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/62/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_if_the_s2i_build_is_finished_as_expected_performing_a_non_native_build_with_persistence_enabled/">features.image.kogito-builder.kogito-builder image tests.Verify if the s2i build is finished as expected performing a non native build with persistence enabled</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/62/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_that_the_Kogito_Quarkus_Serverless_Workflow_Extension_is_building_the_service_properly/">features.image.kogito-builder.kogito-builder image tests.Verify that the Kogito Quarkus Serverless Workflow Extension is building the service properly</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/62/testReport/features.image.kogito-common-builder-jvm/kogito-builder image JVM build tests/Verify_if_the_s2i_build_is_finished_as_expected_with_uber_jar_package_type_built/">features.image.kogito-common-builder-jvm.kogito-builder image JVM build tests.Verify if the s2i build is finished as expected with uber-jar package type built</a></summary>
S2I build failed, check logs!
</details>

**Comment by kie-ci1** [comment]
**PR job** `#64` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/64/display/redirect

**Test results:**
- PASSED: 234
- FAILED: 4

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/64/testReport/features.image.kogito-builder-native/kogito-builder image native build tests/Verify_if_the_s2i_build_is_finished_as_expected_performing_a_native_build_with_persistence_enabled___Step_1:_build_the_application_and_copy_to_the_runtime_image/">features.image.kogito-builder-native.kogito-builder image native build tests.Verify if the s2i build is finished as expected performing a native build with persistence enabled - Step 1: build the application and copy to the runtime image</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/64/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_if_the_s2i_build_is_finished_as_expected_performing_a_non_native_build_with_persistence_enabled/">features.image.kogito-builder.kogito-builder image tests.Verify if the s2i build is finished as expected performing a non native build with persistence enabled</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/64/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_that_the_Kogito_Quarkus_Serverless_Workflow_Extension_is_building_the_service_properly/">features.image.kogito-builder.kogito-builder image tests.Verify that the Kogito Quarkus Serverless Workflow Extension is building the service properly</a></summary>
S2I build failed, check logs!
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/64/testReport/features.image.kogito-common-builder-jvm/kogito-builder image JVM build tests/Verify_if_the_s2i_build_is_finished_as_expected_with_uber_jar_package_type_built/">features.image.kogito-common-builder-jvm.kogito-builder image JVM build tests.Verify if the s2i build is finished as expected with uber-jar package type built</a></summary>
S2I build failed, check logs!
</details>

**Comment by radtriste** [comment]
jenkins retest this

File: RELEASE_NOTES.md (hunk 1 above)
Comment (radtriste) [reply RELEASE_NOTES.md]: @spolti this is not solved. Still showing `77910` :)

File: modules/kogito-s2i-core/module.yaml (hunk 2 above)
Comment (radtriste) [reply modules/kogito-s2i-core/module.yaml]: solved ?

**Comment by kie-ci1** [comment]
**PR job** `#65` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/65/display/redirect

**Test results:**
- PASSED: 237
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/65/testReport/features.image.kogito-builder/kogito-builder image tests/Verify_that_the_Kogito_Quarkus_Serverless_Workflow_Extension_is_building_the_service_properly/">features.image.kogito-builder.kogito-builder image tests.Verify that the Kogito Quarkus Serverless Workflow Extension is building the service properly</a></summary>
S2I build failed, check logs!
</details>

**Comment by kie-ci1** [comment]
**PR job** `#68` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/68/display/redirect

**Test results:**
- PASSED: 237
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/68/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_trusty_postgresql___Build/Test_kogito_trusty_postgresql___Verify_if_the_DEFAULT_MEM_RATIO_properties_are_overridden_with_different_values_from_user_provided_Xmx_and_Xms/">Build & Test Images / Build&Test kogito-trusty-postgresql / Build/Test kogito-trusty-postgresql / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the DEFAULT MEM RATIO properties are overridden with different values from user provided Xmx and Xms</a></summary>
HOOK-ERROR in after_scenario: APIError: 409 Client Error for http+docker://localhost/v1.22/containers/ec841bea05c6d44d3dfc438e28d011d18af4a12adca6633b6066695a533f3309/kill: Conflict ("b'Cannot kill container: ec841bea05c6d44d3dfc438e28d011d18af4a12adca6633b6066695a533f3309: Container ec841bea05c6d44d3dfc438e28d011d18af4a12adca6633b6066695a533f3309 is not running'")
</details>

**Comment by spolti** [comment]
jenkins retest this

**Pull request merged**